exchanges/refocus/src/refocusExchange.ts

```diff
@@ -1,12 +1,36 @@
 import { pipe, tap } from 'wonka';
 import { Exchange, Operation } from '@urql/core';
 
+// Source: https://github.com/vercel/swr/blob/4d6be3df441f925d235a4b26914efcc85ae335a3/src/libs/is-document-visible.ts
+const isDocumentVisible = (): boolean => {
+  if (
+    typeof document !== 'undefined' &&
+    typeof document.visibilityState !== 'undefined'
+  ) {
+    return document.visibilityState !== 'hidden';
+  }
+  // always assume it's visible
+  return true;
+};
+
+// Source: https://github.com/vercel/swr/blob/4d6be3df441f925d235a4b26914efcc85ae335a3/src/libs/is-online.ts
+const isOnline = (): boolean => {
+  if (typeof navigator.onLine !== 'undefined') {
+    return navigator.onLine;
+  }
+  // always assume it's online
+  return true;
+};
```

Comment:
This check should be removed as we have different systems that care about offline; so generally, if we're offline, we should still issue the reexecution and let other exchanges deal with offline/online behaviour so this remains super predictable.

Reply:
Oh, I didn't know that there are other exchanges already dealing with the offline/online. Good to know 👍

Reply:
Specifically the `offlineExchange` as part of Graphcache adds a lot of offline support. But I actually just realised why you added this. I would still assume though that it's not this exchange's responsibility to care about offline. Especially since by the time the `refocusExchange` re-issues the operation and the request starts, the device may go offline.